### ISSUE TITLE
% in the beginning of lines isn't read

### ISSUE THREAD

**Issue by parhamdoustdar** [robertmeta/swiftmac#18]
Steps to reproduce:

1. Open an empty buffer
2. Type `%s-%s`
3. Press `C-e l` to read current line
4. Note that the first % isn't pronounced

**Comment by robertmeta** [comment]
Yep, reproduced on my end... now to find out why.

**Comment by robertmeta** [comment]
@parhamdoustdar when debugging this, I noticed it happens with the default mac server, can you confirm?  That means it is when moving it over to the Voiceover system. 

Still can fix it by doing the replacement with "percent" on my end, but strange.

**Comment by robertmeta** [comment]
By putting the % -> "percent" change in the core replacements, was able to fix it but it is boggling.

**Comment by parhamdoustdar** [comment]
Hi,
I just verified, this does also happen with the Mac speech server.
Could it be that the speech generation API uses `%` as an escape sequence on MacOS? If you just run swiftmac and put in `q {%%s} <RET> d`, the percent sign gets spoken, which makes me think that it’s an escape sequence for something.
I think the solution might be to replace  `%` in the beginning of a string with `%%`. I also verified that something like `q {%%}` just produces “percent”, which means that you don’t have to escape the second percent sign, just the first.
TL;DR:
1. `%s` should become `%%s`
2. `%%` should become `%%%`
I’m also very curious if we can figure out what these escape sequences are. Maybe they will save you some work in the future, or make the code smaller, if you can use these escape sequences?

**Comment by robertmeta** [comment]
@parhamdoustdar it actually gets stranger if you keep adding in more % entries.    It is som something in the voiceover system for sure, because I dug down and made sure the extract strings are going over to voiceover. But, if you have a string like 

% % % % % % % % % % % %

You get an incorrect reported number of percents and it is not off by one.

**Comment by robertmeta** [comment]
But, 1.0.5 should fix it for now until I puzzle out why exactly it happens.  But at least it happens on mac server as well.

**Comment by parhamdoustdar** [comment]
Wow. Now I know what you mean by painfully figuring VoiceOver out.
